src/ctap/customization.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::ctap::data_formats::{CredentialProtectionPolicy, EnterpriseAttestationMode};
+
+/// ###########################################################################
+/// Constants for adjusting privacy and protection levels.
+/// ###########################################################################
+
+/// Changes the default level for the credProtect extension.
+///
+/// You can change this value to one of the following for more privacy:
+/// - CredentialProtectionPolicy::UserVerificationOptionalWithCredentialIdList
+/// - CredentialProtectionPolicy::UserVerificationRequired
+///
+/// UserVerificationOptionalWithCredentialIdList
+/// Resident credentials that are not in the allowList or excludeList can not be
+/// discovered without user verification.
+///
+/// UserVerificationRequired
+/// No resident credentials can be discovered without user verification.
```

Comment:
I think the intent is different:

Here, documentation is about the implications of setting a _default_ level. In `data_formats.rs`, it is about the data structure, that is used in the default level and the per-credential level. The documentation is similar, of course. I added it there.
But, I really want this information to also be in `customization.rs`, as users shouldn't have to go look for the information they need to make an educated choice.

Reply:
Same as in another comment, it looks like you see this file as a linear read for the user. In which case it's not really source code but more instructions to the user to update some constants. In which case maybe you just want to use comments instead of documentation and then freely organize the file (ignoring that it's actually a Rust file and that documentation is code and has a semantics).

Documentation (and comments) is maintenance burden so it's preferable to avoid writing too much and also avoid duplication. If you still prefer to go for documentation instead of comments, I would consider the fact that from IDEs (like emacs, etc) one can easily jump to the definition of an item and thus its documentation. But it is also possible to read the documentation in a browser and click on hyperlinks:

```
cargo doc --document-private-items --features=std
google-chrome target/doc/ctap2/ctap/customization/index.html
```

To create a hyperlink inside documentation you just need to write `See [CredentialProtectionPolicy] for a detailed description of possible values.` but it's also possible in this case to click directly on the type of the constant.